mockgcp/mockcompute/firewallpoliciesv1.go

```diff
@@ -0,0 +1,493 @@
+
+	name, err := s.parseFirewallPolicyName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.FirewallPolicy{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	rules := []*pb.FirewallPolicyRule{}
+	for _, rule := range obj.Rules {
+		if rule.Priority != nil && *rule.Priority == *req.Priority {
+			// update the rule
+			r := req.GetFirewallPolicyRuleResource()
+			r.Priority = PtrTo(*rule.Priority)
+			// RuleTupleCount is output only, calculation of the complexity of a single firewall policy rule.
+			// Manually set different ruleTupleCount to match the realGCP log
+			if r.TargetResources != nil {
+				r.RuleTupleCount = PtrTo(int32(4))
+			} else {
+				r.RuleTupleCount = PtrTo(int32(2))
+			}
+			r.Kind = PtrTo("compute#firewallPolicyRule")
+			if r.Description == nil {
+				r.Description = PtrTo("")
+			}
+			rules = append(rules, r)
+		} else {
+			rules = append(rules, rule)
+		}
+	}
+
+	obj.Rules = rules
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	op := &pb.Operation{
+		TargetId:      obj.Id,
+		TargetLink:    obj.SelfLink,
+		OperationType: PtrTo("patchFirewallRuleInFirewallPolicy"),
+		User:          PtrTo("user@example.com"),
+	}
+	return s.startGlobalOrganizationLRO(ctx, op, func() (proto.Message, error) {
+		return obj, nil
+	})
+}
+func (s *FirewallPoliciesV1) RemoveRule(ctx context.Context, req *pb.RemoveRuleFirewallPolicyRequest) (*pb.Operation, error) {
+	reqName := "locations/global/firewallPolicies/" + req.GetFirewallPolicy()
+	name, err := s.parseFirewallPolicyName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.FirewallPolicy{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	rules := []*pb.FirewallPolicyRule{}
+	for _, rule := range obj.Rules {
+		if rule.Priority != nil && *rule.Priority == *req.Priority {
+			// remove the rule
+			continue
+		} else {
+			rules = append(rules, rule)
+		}
+	}
+
+	obj.Rules = rules
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	op := &pb.Operation{
+		TargetId:      obj.Id,
+		TargetLink:    obj.SelfLink,
+		OperationType: PtrTo("removeFirewallRuleFromFirewallPolicy"),
+		User:          PtrTo("user@example.com"),
+	}
+	return s.startGlobalOrganizationLRO(ctx, op, func() (proto.Message, error) {
+		return obj, nil
+	})
+}
+
+type firewallPolicyName struct {
+	Name string
+}
+
+func (n *firewallPolicyName) String() string {
+	return "locations/global/firewallPolicies/" + n.Name
+}
+
+// parseFirewallPolicyName parses a string into a firewallPolicyName.
+// The expected form is `locations/global/firewallPolicies/*`.
+func (s *MockService) parseFirewallPolicyName(name string) (*firewallPolicyName, error) {
+	tokens := strings.Split(name, "/")
+
+	if len(tokens) == 4 && tokens[2] == "firewallPolicies" {
```

Comment:
Nit: You should probably check tokens[0] and tokens[1] also, just defensive coding